### PR TITLE
2.x and cake 4.x

### PR DESCRIPTION
**Pull request by saeideng**

### CODE REVIEW

File: src/Authenticator/AuthenticatorCollection.php [hunk 1]
```diff
@@ -83,7 +83,7 @@ protected function _resolveClassName($class)
      * @return void
      * @throws \RuntimeException
      */
-    protected function _throwMissingClassError($class, $plugin)
+    protected function _throwMissingClassError($class, $plugin): void
```
Comment (ADmad) [inline src/Authenticator/AuthenticatorCollection.php]: No typehint for arguments?

Comment (saeideng) [reply src/Authenticator/AuthenticatorCollection.php]: for typehints I will open another PR
just `:void` for return type was required by core here :)

Comment (saeideng) [reply src/Authenticator/AuthenticatorCollection.php]: looks I have to do it in this PR , php 7.1 needs it

**Comment by saeideng** [comment]
I pushed these commits into `2.x` after merge

**Pull request merged**